### PR TITLE
Finish the jstd-adapter to be 'really' useful .

### PR DESCRIPTION
**Pull request by tcorral**
Add all the jstd assertions.
Add the behaviour to execute setUp and tearDown callbacks.
Add a node.js script to replace DOC comments from test files.
Add the tests to test that all works properly.

### CODE REVIEW

**Comment by vojtajina** [comment]
Awesome @tcorral !

Can you please revert the changes to style ? So that it is just adding the matchers ?

``` javascript
// change this
function assertTrue () {}

// to
var assertTrue = function() {};
```

File: jstd-adapter.js [hunk 1]
```diff
@@ -11,54 +11,461 @@
...
 
 
-var __jstd_specs = {};
-var TestCase = function(name) {
-  var prototype = __jstd_specs[name] = {};
+var __jstd_specs = {}, startJasmine, assert, assertTrue;
+
+/**
+ * TestCase replacement that will use Jasmine describe and it functions.
+ * @param name
```
Comment (vojtajina) [inline jstd-adapter.js]: I don't think these comments are any helpful, can you remove them ?

File: jstd-adapter.js [hunk 2]
```diff
@@ -11,54 +11,461 @@
...
+}
+
+// helper functions
+function emptyFunc() {
+}
+/**
+ * injectHTML is a function to add content to the body of the page.
+ * @param sHTML
+ */
+function injectHTML(sHTML) {
```
Comment (vojtajina) [inline jstd-adapter.js]: is this actually JsTD function ?

File: jstd-adapter.js [hunk 3]
```diff
@@ -11,54 +11,461 @@
...
+ * @constructor
+ */
+function ConditionalTestCase(name, condition, proto, type) {
+  if (condition()) {
+    return TestCase(name, proto, type);
+  }
+}
+
+// helper functions
+function emptyFunc() {
```
Comment (vojtajina) [inline jstd-adapter.js]: is this jstd function ? cause it's poluting global space, just inline it, it's not a big deal...

File: jstd-adapter.js [hunk 4]
```diff
@@ -11,54 +11,461 @@
...
-        it(specName.replace(/^test/, 'should '), prototype[specName]);
+    // setUp and tearDown callbacks are executed as beforeEach and afterEach
+    describe(testName, function () {
+      beforeEach(function () {
+        setUp.call(this);
+      });
+      afterEach(function () {
+        tearDown.call(this);
+        // Clean document.body content.
+        document.body.innerHTML = '';
```
Comment (vojtajina) [inline jstd-adapter.js:322]: I don't think the adapter should do this - that's responsibility of the tests to clean up the DOM (if they append some stuff to it)

File: karma.conf.js [hunk 5]
```diff
@@ -54,7 +54,7 @@ autoWatch = true;
 // - Safari (only Mac)
 // - PhantomJS
 // - IE (only Windows)
-browsers = ['Chrome'];
+browsers = ["Chrome"];
```
Comment (vojtajina) [inline karma.conf.js]: revert this change please

File: jstd-adapter.js (hunk 4 above)
Comment (tcorral) [inline jstd-adapter.js:322]: If you clone the jstest-driver project code you will be able to see how it works.
`git clone https://code.google.com/p/js-test-driver/`

To explain this you will need to see this two files:

**Path file:**
JsTestDriver\src\com\google\jstestdriver\javascript\plugins\TestRunnerPlugin.js

**Line Number: 188**

**Path file:**
JsTestDriver\JsTestDriver\src\com\google\jstestdriver\coverage\javascript\plugin.js

**Line Number: 22**

In these lines of code you can see how jstestdriver cleans the body after execute all the 
tests, I only try to replicate the same behaviour with:

``` javascript
 document.body.innerHTML = '';
```

File: jstd-adapter.js (hunk 3 above)
Comment (tcorral) [inline jstd-adapter.js]: If you clone the jstest-driver project code you will be able to see how it works.
    git clone https://code.google.com/p/js-test-driver/

_To explain this you will need to see this two files:_

**Path file:**
JsTestDriver\JsTestDriver\src\com\google\jstestdriver\javascript\TestCaseBuilder.js

**Line Number: 115**

In this file you can see the declaration of ConditionalTestCase

*_Path file:_ *
JsTestDriver\JsTestDriver\src\com\google\jstestdriver\javascript\config.js

**Line Number: 46 & 47**

In this other file you can see how ConditionalTestCase is declared as global.

File: jstd-adapter.js (hunk 2 above)
Comment (tcorral) [inline jstd-adapter.js]: Ok I'll remove emptyFunc declaration it was only to improve the size of the file after being minified.

File: jstd-adapter.js (hunk 1 above)
Comment (tcorral) [inline jstd-adapter.js]: Ok, I'll remove all the commented code.

File: karma.conf.js (hunk 5 above)
Comment (tcorral) [inline karma.conf.js]: Sure!

**Comment by vojtajina** [comment]
In order to add this plugin to karma-runner org, it should follow the style guide (https://github.com/karma-runner/karma/blob/master/CONTRIBUTING.md).

``` javascript
var a, b;
a = true;
b = function() {};

// should be
var a = true;
var b = function() {};
```

Use two spaces for indentation.

Use single quotes.

Then, we should add a grunt task for building it (wrap into closure) and explicitly expose public API onto window. Also separating matchers into a standalone file. I can take care of this step, once we merge this PR.

Btw. is the "html in commments" feature part of JsTD ? Cause I don't remember that. If so, we might make it a preprocessor. Even though, I really don't understand why you need to put this stuff into comments, rather than calling some API like `injectHTML` in your case...

**Comment by tcorral** [comment]
HTML in comments is a feature of JsTD 
https://code.google.com/p/js-test-driver/wiki/HtmlDoc , I thing that this feature is the worst way to inject HTMLinside the document but people uses it and we should take care about it, this is the reason I wrote the replaceDOC to preprocess the files before launch the tests and created the injectHTML function.
I'll commit the changes in styles in 5 minutes. :)

**Comment by tcorral** [comment]
Please could you finish the grunt task? I'm pretty sure you will do it better than me. :)

**Comment by vojtajina** [comment]
@tcorral Awesome ! Yep, I will take care of the grunt tasks.

Last thing, can you squash these changes into two commits:
`feat: add missing matchers`
`feat: add script for replacing DOC comments`

Btw, I didn't know about this DOC comments.

Again, thanks a lot @tcorral !!!

**Comment by tcorral** [comment]
Ok. I'll do two new pull requests one per each commit you propose.
